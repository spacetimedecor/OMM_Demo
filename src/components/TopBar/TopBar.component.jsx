import StyledTopBar from "./TopBar.styled";
import { OverlayMenu } from "../../components/OverlayMenu/OverlayMenu.component";
import { StyledImage } from "../common";
import PropTypes from "prop-types";
import {default as React, useState, memo} from 'react';
import logo from "../../assets/omm_logo_black.webp";
import burger from "../../assets/hamburger-menu.svg";

const TopBar = (props) => {
  const [showOverlay, setShowOverlay] = useState(false);

  const handleMenu = () => setShowOverlay(!showOverlay);

  return (
    <>
      <StyledTopBar {...props}>
        <StyledImage
          src={logo}
          alt="logo"
        />
        <StyledImage
          src={burger}
          alt="menu"
          onClick={handleMenu}
        />
        { showOverlay && <OverlayMenu onClick={handleMenu} /> }
      </StyledTopBar>
    </>
  );
}

export default memo(TopBar);

TopBar.propTypes = {
  /**
   * Padding around the graphics
   */
  imagePadding: PropTypes.string,

  /**
   * Height
   */
  height: PropTypes.string,

  /**
   * Background color
   */
  backgroundColor: PropTypes.string,

  /**
   * Border color
   */
  borderColor: PropTypes.string,

  /**
   * Color
   */
  color: PropTypes.string,
}
