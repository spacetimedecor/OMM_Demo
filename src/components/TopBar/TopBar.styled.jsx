import styled from "styled-components";

const StyledTopBar = styled("nav")`
  height: ${({ height, theme }) => height || theme.layout.topBar.height };
  background: ${({ backgroundColor, theme }) => backgroundColor || theme.colors.backgroundColor1 };
  color: ${({ color, theme }) => color || theme.colors.primaryColor };
  display: flex;
  justify-content: space-between;
  position: fixed;
  width: 100%;
  z-index: 50;

  ${ ({ withBorder, borderColor, theme }) => withBorder ? 
    `${ borderColor || theme.colors.borderColor } 1px solid` : 
    '' 
  }
`;

export default StyledTopBar;
