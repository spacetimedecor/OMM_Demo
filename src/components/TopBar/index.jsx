export { default } from "./TopBar.component";
