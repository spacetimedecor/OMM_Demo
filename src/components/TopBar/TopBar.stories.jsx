import TopBar from "./TopBar.component";

import React from "react";

export default {
    title: 'Components/TopBar',
    component: TopBar,
    parameters: {
        layout: 'fullscreen',
    },
    argTypes: {
        backgroundColor: { control: 'color' },
        borderColor: { control: 'color' },
        color: { control: 'color' },
        height: { control: 'text' },
        imagePadding: { control: 'text' },
        withBorder: { control: 'boolean' }
    }
};

const Template = (args) => <TopBar {...args} />;

export const Primary = Template.bind({});
Primary.args = {
    primary: true,
};
