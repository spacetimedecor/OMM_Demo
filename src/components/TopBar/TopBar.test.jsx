import * as React from "react";

import themed from "../../theme/themed";
import TopBar from "./TopBar.component";

describe("TopBar", () => {
  it("should render correctly", () => {
    const component = themed(
      <TopBar>
        Test
      </TopBar>,
    ).toJSON();

    expect(component).toMatchSnapshot();
  });
});
