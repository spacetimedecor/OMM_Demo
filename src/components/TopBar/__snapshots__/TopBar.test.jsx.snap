// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar should render correctly 1`] = `
<nav
  className="sc-bdvvtL figgNX"
>
  <img
    alt="logo"
    className="sc-gsDKAQ lfkiim"
    src=""
  />
  <img
    alt="menu"
    className="sc-gsDKAQ eMyjrx"
    onClick={[Function]}
    src=""
  />
</nav>
`;
